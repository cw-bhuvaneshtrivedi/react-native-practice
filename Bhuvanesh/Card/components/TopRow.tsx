import React from "react";
import { View, Image, Text } from "react-native";
import { Entypo } from "@expo/vector-icons";

import globalStyles from "../Styles";

type TopRowProps = {
  makeName: string;
  modelName: string;
  imageUrl: string;
};

export const TopRow = ({ makeName, modelName, imageUrl }: TopRowProps) => {
  const { flexRow, spaceBetween, image, mkName, mdName, topRow } = globalStyles;
  return (
    <View style={[flexRow, spaceBetween, topRow]}>
      {/* <Image style={image} source={require("../assets/icon.png")} /> */}
      <Image style={image} source={{ uri: imageUrl }} testID="image" />
      <View>
        <Text style={mkName} testID="mkName">
          {makeName}
        </Text>
        <Text style={mdName} testID="mdName">
          {modelName}
        </Text>
      </View>
      <Entypo name="chevron-right" size={24} color="#6f6f6f" />
    </View>
  );
};

// TopRow.defaultProps = {
//   makeName: "Mahindra",
//   modelName: "Thar",
//   imageUrl:
//     "https://imgd.aeplcdn.com/310x174/n/cw/ec/130591/fronx-exterior-right-front-three-quarter-109.jpeg?isig=0&q=80",
// };
