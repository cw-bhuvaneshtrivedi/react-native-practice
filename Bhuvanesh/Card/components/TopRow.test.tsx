import { render } from "@testing-library/react-native";
import { TopRow } from "./TopRow";

type TopRowProps = {
  makeName: string;
  modelName: string;
  imageUrl: string;
};
describe("Top Row ", () => {
  const mkName = "Mahindra";
  const mdName = "Thar";
  const imageUrl =
    "https://imgd.aeplcdn.com/310x174/n/cw/ec/130591/fronx-exterior-right-front-three-quarter-109.jpeg?isig=0&q=80";

  it("Is image Visible", () => {
    const { getByTestId } = render(
      <TopRow imageUrl={imageUrl} makeName={mkName} modelName={mdName} />
    );
    const image = getByTestId("image");
    expect(image.props.source.uri).toEqual(imageUrl);
  });

  it("Is MakeName having correct color", () => {
    const { getByTestId } = render(
      <TopRow imageUrl={imageUrl} makeName={mkName} modelName={mdName} />
    );
    const make = getByTestId("mkName");
    expect(make.props.style.color).toEqual("#aaaaaa");
  });

  it("Is Model Visible and has correct text", () => {
    const { getByTestId } = render(
      <TopRow imageUrl={imageUrl} makeName={mkName} modelName={mdName} />
    );
    const model = getByTestId("mdName");
    expect(model.props.children).toEqual("Thar");
  });

  it("should match snapshot", () => {
    const snap = render(
      <TopRow makeName={mkName} modelName={mdName} imageUrl={imageUrl} />
    );
    expect(snap).toMatchSnapshot();
  });
});
