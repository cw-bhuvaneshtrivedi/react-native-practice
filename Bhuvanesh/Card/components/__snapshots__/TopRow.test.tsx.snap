// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Row  should match snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
      {
        "alignItems": "center",
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
      },
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://imgd.aeplcdn.com/310x174/n/cw/ec/130591/fronx-exterior-right-front-three-quarter-109.jpeg?isig=0&q=80",
      }
    }
    style={
      {
        "height": 73,
        "width": 130,
      }
    }
    testID="image"
  />
  <View>
    <Text
      style={
        {
          "color": "#aaaaaa",
          "fontSize": 13,
        }
      }
      testID="mkName"
    >
      Mahindra
    </Text>
    <Text
      style={
        {
          "color": "#484848",
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
      testID="mdName"
    >
      Thar
    </Text>
  </View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#6f6f6f",
          "fontSize": 24,
        },
        undefined,
        {
          "fontFamily": "entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ï…ƒ
  </Text>
</View>
`;
